packages/client/src/usePromiseValue.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState, useRef } from "react";
```

Comment:
this also seems useful and general enough to be put in MUD (maybe the new react package, or maybe std-client or utils)